--- a/src/gallium/targets/teflon/meson.build
+++ b/src/gallium/targets/teflon/meson.build
@@ -1,3 +1,14 @@
+teflon_deps = [idep_nir, idep_mesautil]
+if with_gallium_etnaviv
+  teflon_deps += driver_etnaviv
+endif
+if with_gallium_rocket
+  teflon_deps += driver_rocket
+endif
+if with_gallium_ethosu
+  teflon_deps += driver_ethosu
+endif
+
 libteflon = shared_library(
   'teflon',
   [ 'teflon_target.c' ],
@@ -6,13 +17,7 @@ libteflon = shared_library(
   link_with : [libpipe_loader_static, libws_null, libwsw, libswdri, libswkmsdri, libgallium ],
   gnu_symbol_visibility : 'hidden',
   link_args : ld_args_build_id,
-  dependencies : [
-    driver_etnaviv,
-    driver_rocket,
-    driver_ethosu,
-    idep_nir,
-    idep_mesautil,
-  ],
+  dependencies : teflon_deps,
   install : true,
 )
 
