--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -337,6 +337,7 @@ option(FORCE_INTERNAL_BZIP2 "Use interna
 option(FORCE_INTERNAL_ASMJIT "Use internal asmjit" ON)
 mark_as_advanced( FORCE_INTERNAL_ASMJIT )
 
+if( CMAKE_CROSSCOMPILING )
 if (HAVE_VULKAN)
 	add_subdirectory( libraries/ZVulkan )
 endif()
@@ -375,6 +376,7 @@ if( ${HAVE_VM_JIT} )
 		set( ASMJIT_LIBRARY asmjit )
 	endif()
 endif()
+endif()
 
 if( BZIP2_FOUND AND NOT FORCE_INTERNAL_BZIP2 )
 	message( STATUS "Using system bzip2 library, includes found at ${BZIP2_INCLUDE_DIR}" )
@@ -409,12 +411,14 @@ option( DYN_OPENAL "Dynamically load Ope
 add_subdirectory( libraries/lzma )
 add_subdirectory( libraries/miniz )
 add_subdirectory( tools )
+if( CMAKE_CROSSCOMPILING )
 add_subdirectory( wadsrc )
 add_subdirectory( wadsrc_bm )
 add_subdirectory( wadsrc_lights )
 add_subdirectory( wadsrc_extra )
 add_subdirectory( wadsrc_widepix )
 add_subdirectory( src )
+endif()
 
 if( NOT CMAKE_CROSSCOMPILING )
 	export(TARGETS ${CROSS_EXPORTS} FILE "${CMAKE_BINARY_DIR}/ImportExecutables.cmake" )
