--- a/src/common/engine/i_interface.cpp
+++ b/src/common/engine/i_interface.cpp
@@ -6,13 +6,13 @@
 #include "gstrings.h"
 #include "version.h"
 
-static_assert(sizeof(void*) == 8,
-	"Only LP64/LLP64 builds are officially supported. "
-	"Please do not attempt to build for other platforms; "
-	"even if the program succeeds in a MAP01 smoke test, "
-	"there are e.g. known visual artifacts "
-	"<https://forum.zdoom.org/viewtopic.php?f=7&t=75673> "
-	"that lead to a bad user experience.");
+//static_assert(sizeof(void*) == 8,
+//	"Only LP64/LLP64 builds are officially supported. "
+//	"Please do not attempt to build for other platforms; "
+//	"even if the program succeeds in a MAP01 smoke test, "
+//	"there are e.g. known visual artifacts "
+//	"<https://forum.zdoom.org/viewtopic.php?f=7&t=75673> "
+//	"that lead to a bad user experience.");
 
 // Some global engine variables taken out of the backend code.
 FStartupScreen* StartWindow;
