diff -ruN qttools-opensource-src-5.8.0.orig/src/kmap2qmap/main.cpp qttools-opensource-src-5.8.0/src/kmap2qmap/main.cpp
--- qttools-opensource-src-5.8.0.orig/src/kmap2qmap/main.cpp	2017-01-02 07:50:13.000000000 +0100
+++ qttools-opensource-src-5.8.0/src/kmap2qmap/main.cpp	2017-02-21 03:31:59.598892309 +0100
@@ -270,6 +270,8 @@
     { "Pause",         Qt::Key_Pause },
     { "VolumeDown",    Qt::Key_VolumeDown },
     { "VolumeUp",      Qt::Key_VolumeUp },
+    { "VolumeMute",    Qt::Key_VolumeMute },
+    { "MicMute",       Qt::Key_MicMute },
     { "MediaTogglePlayPause", Qt::Key_MediaTogglePlayPause },
     { "MediaPlay",     Qt::Key_MediaPlay },
     { "MediaPause",    Qt::Key_MediaPause },
@@ -286,6 +288,12 @@
     { "ToggleCallHangup", Qt::Key_ToggleCallHangup },
     { "VoiceDial"       , Qt::Key_VoiceDial },
     { "LastNumberRedial", Qt::Key_LastNumberRedial },
+    { "Phone",            Qt::Key_Phone },
+    { "HomePage",         Qt::Key_HomePage },
+    { "Menu",             Qt::Key_Menu },
+    { "Back",             Qt::Key_Back },
+    { "LaunchMail",       Qt::Key_LaunchMail },
+    { "LaunchMedia",      Qt::Key_LaunchMedia },
 
     { "KP_0",        Qt::Key_0 | Qt::KeypadModifier },
     { "KP_1",        Qt::Key_1 | Qt::KeypadModifier },
