--- a/src/platformsupport/platformsupport.pro
+++ b/src/platformsupport/platformsupport.pro
@@ -6,13 +6,13 @@ SUBDIRS = \
     eventdispatchers \
     devicediscovery \
     fbconvenience \
-    themes
+    themes \
+    input
 
 qtConfig(freetype)|darwin|win32: \
     SUBDIRS += fontdatabases
 
 qtConfig(evdev)|qtConfig(tslib)|qtConfig(libinput)|qtConfig(integrityhid)|qtConfig(xkbcommon) {
-    SUBDIRS += input
     input.depends += devicediscovery
 }
 
