--- a/src/corelib/kernel/qcrashhandler.cpp
+++ b/src/corelib/kernel/qcrashhandler.cpp
@@ -62,7 +62,7 @@ QT_BEGIN_NAMESPACE
 
 QtCrashHandler QSegfaultHandler::callback = 0;
 
-#if defined(__GLIBC__) && (__GLIBC__ >= 2) && !defined(__UCLIBC__) && !defined(QT_LINUXBASE)
+#if defined(__GLIBC__) && (__GLIBC__ >= 2) && (!defined(__UCLIBC__) || (defined(__UCLIBC__) && defined(UCLIBC_HAS_BACKTRACE))) && !defined(QT_LINUXBASE)
 QT_BEGIN_INCLUDE_NAMESPACE
 # include "qstring.h"
 # include <execinfo.h>
