--- a/Source/cmake/OptionsCommon.cmake
+++ b/Source/cmake/OptionsCommon.cmake
@@ -20,17 +20,17 @@ if (WTF_CPU_ARM)
     int main() {}
     ")
 
-    if (COMPILER_IS_CLANG AND NOT (${CMAKE_SYSTEM_NAME} STREQUAL "Darwin"))
-        set(CLANG_EXTRA_ARM_ARGS " -mthumb")
+    if (NOT (${CMAKE_SYSTEM_NAME} STREQUAL "Darwin"))
+        set(EXTRA_ARM_ARGS " -mthumb")
     endif ()
 
-    set(CMAKE_REQUIRED_FLAGS "${CLANG_EXTRA_ARM_ARGS}")
+    set(CMAKE_REQUIRED_FLAGS "${EXTRA_ARM_ARGS}")
     CHECK_CXX_SOURCE_COMPILES("${ARM_THUMB2_TEST_SOURCE}" ARM_THUMB2_DETECTED)
     unset(CMAKE_REQUIRED_FLAGS)
 
     if (ARM_THUMB2_DETECTED AND NOT (${CMAKE_SYSTEM_NAME} STREQUAL "Darwin"))
-        string(APPEND CMAKE_C_FLAGS " ${CLANG_EXTRA_ARM_ARGS}")
-        string(APPEND CMAKE_CXX_FLAGS " ${CLANG_EXTRA_ARM_ARGS}")
+        string(APPEND CMAKE_C_FLAGS " ${EXTRA_ARM_ARGS}")
+        string(APPEND CMAKE_CXX_FLAGS " ${EXTRA_ARM_ARGS}")
     endif ()
 endif ()
 
